src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPathContextProvider.cs

```diff
@@ -204,44 +180,24 @@ public bool TryCreateSolutionContext(string solutionDirectory, out IVsPathContex
             return context;
         }
 
-        private IVsPathContext GetPathContextForProjectJson(
-            IVsProjectAdapter vsProjectAdapter)
+        public IVsPathContext GetSolutionPathContext()
         {
-            // generate project.lock.json file path from project file
-            var projectFilePath = vsProjectAdapter.FullProjectPath;
+            return new VsPathContext(NuGetPathContext.Create(_settings.Value));
+        }
 
-            if (!string.IsNullOrEmpty(projectFilePath))
+        private async Task<IVsPathContext> GetPathContextFromAssetsFileAsync(
+            NuGetProject nuGetProject, CancellationToken token)
+        {
+            // It's possible that this project isn't a build integrated NuGet project at all. That is, this project may
```

Comment:
Make this method take a BuildIntegrated project and do the check outside.

Reply:
it wont make any difference, but rather clutter `CreatePathContextAsync` api. I'll prefer to keep it this way which is more readable and easy to understand. Besides, it's the same pattern we had before...

Reply:
Why would call a method `GetPathContextFromAssetsFileAsync` and call it with a project that doesn't have an assets file? 
That's the beef I have. 

I'm saying that the old pattern was bad, and now it's a good time to fix it :)